src/lib/y2storage/proposal/devices_planner.rb

```diff
@@ -173,23 +173,32 @@ def root_device
           else
             root_vol.max_size
           end
-        adjust_btrfs_sizes!(root_vol)
-        init_btrfs_subvolumes!(root_vol)
+        setup_btrfs(root_vol)
         root_vol
       end
 
-      # Sizes have to be adjusted only when using snapshots
-      def adjust_btrfs_sizes!(planned_device)
-        return if !planned_device.btrfs? || !settings.use_snapshots
+      def setup_btrfs(planned_vol)
+        return unless planned_vol.btrfs?
+
+        if settings.use_snapshots && planned_vol.root?
+          log.info "Enabling snapshots for '/'"
+          planned_vol.snapshots = true
+          adjust_btrfs_sizes(planned_vol)
+        end
 
-        log.info "Increasing root filesystem size for snapshots"
+        init_btrfs_subvolumes(planned_vol)
+      end
+
+      # Sizes have to be adjusted only when using snapshots
+      def adjust_btrfs_sizes(planned_device)
         multiplicator = 1.0 + settings.btrfs_increase_percentage / 100.0
+        log.info "Increasing root filesystem size for snapshots (#{multiplicator})"
         planned_device.min_size *= multiplicator
         planned_device.max_size *= multiplicator
       end
 
-      def init_btrfs_subvolumes!(planned_device)
-        return unless planned_device.btrfs? && settings.subvolumes
+      def init_btrfs_subvolumes(planned_device)
+        return unless settings.subvolumes
```

Comment:
I have doubts about we should set the default subvolume path even when the list of subvolumes is empty.

Reply:
Good point... but not for this PR :wink: